lib/ansible/cli/inventory.py

```diff
@@ -193,6 +197,20 @@ def dump(self, stuff):
             import yaml
             from ansible.parsing.yaml.dumper import AnsibleDumper
             results = yaml.dump(stuff, Dumper=AnsibleDumper, default_flow_style=False)
+        elif self.options.toml:
+            try:
+                import toml
+            except ImportError:
+                raise AnsibleError(
+                    'The python "toml" library is required when using the TOML output format'
+                )
+            else:
+                from ansible.plugins.inventory.toml import AnsibleTomlEncoder, convert_yaml_objects_to_native
+
+            if AnsibleTomlEncoder:
+                results = toml.dumps(stuff, encoder=AnsibleTomlEncoder())
```

Comment:
After looking closer, I believe you should encapsulate `toml.dumps` as `toml_dumps` right inside `ansible.plugins.inventory.toml`, because you're duplicating import checks in here.

Example: https://github.com/ansible/ansibullbot/blob/master/ansibullbot/_pickle_compat.py#L7-L10

This whole snippet could be reduced to just:
```python
elif self.options.toml:
    try:
        from ansible.plugins.inventory.toml import toml_dumps
    except ImportError:
        raise AnsibleError(
            'The python "toml" library is required when using the TOML output format'
        )
    results = toml_dumps(stuff)  # it should take care of shimming loader internally, that logic doesn't belong here
```